pkg/k8s/apis/cilium.io/utils/utils.go

```diff
@@ -322,7 +322,7 @@ func namespacesAreValid(namespace string, userNamespaces []string) bool {
 func ParseToCiliumRule(logger *slog.Logger, namespace, name string, uid types.UID, r *api.Rule) *api.Rule {
 	retRule := &api.Rule{}
 	if r.EndpointSelector.LabelSelector != nil {
-		retRule.EndpointSelector = api.NewESFromK8sLabelSelector("", r.EndpointSelector.LabelSelector)
+		retRule.EndpointSelector = api.NewESFromK8sLabelSelector(labels.LabelSourceK8sKeyPrefix, r.EndpointSelector.LabelSelector)
```

Comment:
It looks like we've allowed embedding the source prefix in CNPs et al, colon-separated. We should continue to allow this, only adding a source prefix of `k8s` or `any` when none is supplied.

This is also different for NodeSelectors -- we have to see what source is included with nodes.

Reply:
Fixed.